ci_framework/roles/build_containers/tasks/main.yml

```diff
@@ -0,0 +1,47 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Ensure containers working directory exists
+  file:
+    path: "{{ cifmw_build_containers_basedir }}"
+    state: directory
+
+- name: Ensure logs directory exists
+  file:
+    path: '{{ cifmw_build_containers_basedir }}/logs'
+    state: directory
+
+- name: Install tcib
+  include_tasks: install.yaml
+
+- name: Generate container_build.sh script
+  template:
+    src: templates/build_containers.sh.j2
+    dest: "{{ cifmw_build_containers_basedir }}/build_containers.sh"
+    mode: 0777
+    force: true
+
+- name: Run container_build.sh script
+  become: true
+  args:
+    chdir: "{{ cifmw_build_containers_basedir }}"
+  ansible.builtin.shell: |
+    set -o pipefail && bash build_containers.sh 2>&1 {{ cifmw_build_containers_timestamper_cmd }} > {{ cifmw_build_containers_basedir }}/logs/build.log
+  environment:
+   PATH: /usr/local/bin:{{ ansible_env.PATH }}
```

Comment:
maybe consuming `{{ cifmw_path }}` would be better - if the /usr/local/bin is missing, we can add it globally.

Though that location is used because of the `become: true` during the install step.